csharp/rocketmq-client-csharp/SimpleConsumer.cs

```diff
@@ -196,12 +196,13 @@ internal override void OnSettingsReceived(rmq.Settings settings)
             }
         }
 
-        public async Task<List<Message>> Receive(int batchSize, TimeSpan invisibleDuration)
+        public async Task<List<Message>> Receive(int batchSize, TimeSpan invisibleDuration, TimeSpan? awaitDuration = null)
         {
             var messageQueue = NextQueue();
             if (null == messageQueue)
             {
-                Logger.Debug("NextQueue returned null");
+                Logger.Debug("NextQueue returned null, waiting for 50ms");
+                await Task.Delay(50);
```

Comment:
Why did we need to delay here?

Reply:
If we start the consumer without any topics. a receive loop became a busy loop with no yield point. this will saturate a client thread and takes up server resources.

Reply:
@colprog If the queue could not be chosen, an exception is preferred rather than an empty message list. Exception make developers could be aware of the unexcepted occation and it is within their discretion. On the other hand, consumer need to do some precheck before startup, including the subscription is set or not.

Reply:
Making Start() lazy is probably out of scope for this changes.
Is throwing an exception here ok with you?

Reply:
> Making Start() lazy is probably out of scope for this changes. Is throwing an exception here ok with you?

Yes, I agree with it!

Reply:
fixed in latest revision